relion/protocols/protocol_subtract.py

```diff
@@ -102,41 +150,133 @@ def _defineParams(self, form):
+                           'model is inadequate at the lowest resolution. '
+                           'Still, in general using higher amplitude contrast '
+                           'on the CTFs (e.g. 10-20%) often yields better '
+                           'results. Therefore, this option is not generally '
+                           'recommended.')
+
         form.addParallelSection(threads=0, mpi=1)
     
     # -------------------------- INSERT steps functions -----------------------
     def _insertAllSteps(self):
+        self.isRelionInput = self.relionInput.get()
         self._initialize()
 
-        if not self.useAll:
+        if not self.useAll or not self.isRelionInput:
             self._insertFunctionStep('convertInputStep')
-
         self._insertFunctionStep('subtractStep')
         self._insertFunctionStep('createOutputStep')
     
     # -------------------------- STEPS functions ------------------------------
     def convertInputStep(self):
         """ Write the input images as a Relion star file. """
+        if not self.isRelionInput:
+            self.inputParticles = self.inputParticlesAll
         imgSet = self.inputParticles.get()
+
         convert.writeSetOfParticles(
             imgSet, self._getFileName('input_star'),
             outputDir=self._getExtraPath(), alignType=ALIGN_PROJ)
-    
+
     def subtractStep(self):
+        if self.isRelionInput:
+            self.subtractStepRelion()
+        else:
+            self.subtractStepNoRelion()
+
+    def subtractStepNoRelion(self):
+        self.numberOfMpi = Integer(1)
+        volume = self.inputVolume.get()
+        volFn = convert.convertBinaryVol(volume,
+                                         self._getExtraPath())
+        params = ' --i %s --subtract_exp --angpix %0.3f' % (volFn,
+                                                            volume.getSamplingRate())
+
+        if self.refMask.hasValue():
+            tmp = self._getTmpPath()
+            newDim = self._getInputParticles().getXDim()
+            newPix = self._getInputParticles().getSamplingRate()
+
+            maskFn = convert.convertMaskThreshold(self.refMask.get(),
+                                                  tmp,
+                                                  newPix,
+                                                  newDim,
+                                                  self.invertMask)
+            params += ' --mask %s' % maskFn
+
+        if self._getInputParticles().isPhaseFlipped():
+            params += ' --ctf_phase_flip'
+
+        if self.doCTF:
+            params += ' --ctf'
+            if self.ignoreCTFUntilFirstPeak:
+                params += ' --ctf_intact_first_peak'
+
+        params += ' --ang %s  --o %s ' % (
+            self._getFileName('input_star'),
+            self._getFileName('output_star').replace(".star", ""))
+
+        try:
+            self.runJob('relion_project', params)
+        except Exception as ex:
```

Comment:
we do not use try/except for such cases. The check should be done in createOutputStep

Reply:
done